components/core/src/package/ident.rs

```diff
@@ -33,7 +33,7 @@ pub trait Identifiable: fmt::Display + Into<PackageIdent> {
     fn fully_qualified(&self) -> bool { self.version().is_some() && self.release().is_some() }
 
     fn valid(&self) -> bool {
-        let re = Regex::new(r"^[A-Za-z0-9_-]+$").unwrap();
+        let re = Regex::new(r"^[A-Za-z0-9_\.-]+$").unwrap();
```

Comment:
Do we want to allow a period at the beginning? At the end? More than one in a row?

Reply:
yeah I think that is fine. It might be odd but it does not raise any problems on our end and it would only add complexity to include such guards.

Reply:
It might be good to add some tests that show names like "...." (:scream:) not breaking everything.

Reply:
Yep agreed, and we need to include Builder tests as well so it would be great if those could be added as well in a separate PR please.  This is a potentially a fairly large change with a potential to be disruptive.  I'd rather open the door slowly, and not allow arbitrary periods in the name - not sure what is the point of a "..." name - this seems terrible to me.

Reply:
ok thats fair. we'll just allow periods in the middle